testing/time.ts

```diff
@@ -157,6 +157,16 @@ function* timerIdGen() {
   while (true) yield i++;
 }
 
+function nextDueNode(): DueNode | null {
+  for (;;) {
+    const dueNode = dueTree.min();
+    if (!dueNode) return null;
+    const hasTimer = dueNode.timers.some((timer) => dueNodes.has(timer.id));
+    if (hasTimer) return dueNode;
+    dueTree.remove(dueNode);
```

Comment:
Removing dueTree item here feels confusing to me. How about fixing `fakeClearTimeout` and `fakeClearInterval` instead?

```diff
 function fakeClearTimeout(id?: number) {
   if (!time) throw new TimeError("no fake time");
-  if (typeof id === "number" && dueNodes.has(id)) {
-    dueNodes.delete(id);
+  if (typeof id === "number") {
+    const dueNode = dueNodes.get(id);
+    if (dueNode) {
+      dueNodes.delete(id);
+      dueTree.remove(dueNode);
+    }
   }
 }
```

Reply:
I did this because I wanted to define the deletion process in one place.

And the code example you provided is incomplete.
`DueNode.timers` contains timers with other ids, so it should not simply be removed from the dueTree.
I added a test about it.

Reply:
hmm, we already do similar deletion in `now` setter. Now I think it's fine to delete tree item here.